### DISCUSSION TITLE
Flex decoder GET option ducumentation

### DISCUSSION THREAD

**Discussion by avicarmeli** [merbanan/rtl_433#2223] [Q&A] [answered]
Hi,
I saw some [conf ](https://github.com/merbanan/rtl_433/blob/master/conf/SMC5326-Remote.conf)files containing the flex get option.
I looked in the [documentation ](https://triq.org/rtl_433/OPERATION.html#decoders) but could not find any instructions on how to use it.
Is there other place to find that information?

**Comment by zuckschwerdt** [comment]
I don't think we have documentation for that.

Example: `get=battery_ok:@4:{1}:[0:Ok 1:Empty]`
It's colon separated keys:
- name (`battery_ok`)
- at which bit to start (`@4`)
- how many bits to read (`{1}`)
- optional mapping (map `0` to `Full`, map `1` to `Empty`)

**Comment by avicarmeli** [comment]
Great. 
So what does the %x stands for in:
`get=src:@8:{16}:%x,`
and how about little/big endians?

**Comment by zuckschwerdt** [comment, marked as answer]
A key starting with `%` is a format specifier for the KV output, in practice it will only ever be something like `%x`, `%X`, `%04x`.
Getters are pretty low level for output, there is not endianess support.
